src/SignalR/clients/cpp/src/signalrclient/connection_impl.cpp

```diff
@@ -39,7 +38,7 @@ namespace signalr
         std::unique_ptr<web_request_factory> web_request_factory, std::unique_ptr<transport_factory> transport_factory)
         : m_base_url(url), m_query_string(query_string), m_connection_state(connection_state::disconnected), m_logger(log_writer, trace_level),
         m_transport(nullptr), m_web_request_factory(std::move(web_request_factory)), m_transport_factory(std::move(transport_factory)),
-        m_message_received([](const utility::string_t&){}), m_disconnected([](){})
+        m_message_received([](const utility::string_t&) noexcept {}), m_disconnected([]() noexcept {})
```

Comment:
Is this the result of following [this](https://docs.microsoft.com/en-us/visualstudio/code-quality/using-the-cpp-core-guidelines-checkers?view=vs-2017) guide?

Are we basically putting noexcept on every method/ctor/callback we can? This wouldn't virally force all callbacks to use the noexcept specifier, right? I hope adding noexcept is never breaking.

Reply:
Yes.

This would not force callbacks to use noexcept.